ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -0,0 +1,592 @@
+   */
+  public GetBlobOperationTest()
+      throws Exception {
+    // Defaults. Tests may override these and do new puts as appropriate.
+    MAX_CHUNK_SIZE = random.nextInt(1024 * 1024) + 1;
+    // a blob size that is greater than the MAX_CHUNK_SIZE and is not a multiple of it. Will result in a composite blob.
+    blobSize = MAX_CHUNK_SIZE * random.nextInt(10) + random.nextInt(MAX_CHUNK_SIZE - 1) + 1;
+    mockSelectorState.set(MockSelectorState.Good);
+    VerifiableProperties vprops = new VerifiableProperties(getDefaultNonBlockingRouterProperties());
+    routerConfig = new RouterConfig(vprops);
+    mockClusterMap = new MockClusterMap();
+    blobIdFactory = new BlobIdFactory(mockClusterMap);
+    routerMetrics = new NonBlockingRouterMetrics(mockClusterMap.getMetricRegistry());
+    mockServerLayout = new MockServerLayout(mockClusterMap);
+    REPLICAS_COUNT = mockClusterMap.getWritablePartitionIds().get(0).getReplicaIds().size();
+    responseHandler = new ResponseHandler(mockClusterMap);
+    MockNetworkClientFactory networkClientFactory =
+        new MockNetworkClientFactory(vprops, mockSelectorState, MAX_PORTS_PLAIN_TEXT, MAX_PORTS_SSL,
+            CHECKOUT_TIMEOUT_MS, mockServerLayout, time);
+    router =
+        new NonBlockingRouter(routerConfig, new NonBlockingRouterMetrics(new MetricRegistry()), networkClientFactory,
+            new LoggingNotificationSystem(), mockClusterMap, time);
+    networkClient = networkClientFactory.getNetworkClient();
+  }
+
+  private void doPut()
+      throws Exception {
+    BlobProperties blobProperties =
+        new BlobProperties(blobSize, "serviceId", "memberId", "contentType", false, Utils.Infinite_Time);
+    byte[] userMetadata = new byte[10];
+    random.nextBytes(userMetadata);
+    putContent = new byte[blobSize];
+    random.nextBytes(putContent);
+    ReadableStreamChannel putChannel = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(putContent));
+    blobIdStr = router.putBlob(blobProperties, userMetadata, putChannel).get();
+  }
+
+  /**
+   * Test {@link GetBlobOperation} instantiation and validate the get methods.
+   * @throws Exception
+   */
+  @Test
+  public void testInstantiation()
+      throws Exception {
+    Callback<ReadableStreamChannel> operationCallback = new Callback<ReadableStreamChannel>() {
+      @Override
+      public void onCompletion(ReadableStreamChannel result, Exception exception) {
+        // no op.
+      }
+    };
+
+    // test a bad case
+    try {
+      new GetBlobOperation(routerConfig, routerMetrics, mockClusterMap, responseHandler, "invalid_id", operationFuture,
+          operationCallback, operationCompleteCallback, blobIdFactory, time);
+      Assert.fail("Instantiation of GetBlobOperation with an invalid blob id must fail");
+    } catch (RouterException e) {
+      Assert.assertEquals("Unexpected exception received on creating GetBlobOperation", RouterErrorCode.InvalidBlobId,
+          e.getErrorCode());
+    }
+
+    doPut();
```

Comment:
In this particular case, yes. Seemed simpler to just call this method at the time. Will instead create the blobId directly if that is more intuitive.